mautic/mautic
# Contact activity events api

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Related user documentation PR URL | /
| Related developer documentation PR URL | https://github.com/mautic/developer-documentation/pull/67
| Issues addressed (#s or URLs) | /
| BC breaks? | It makes timeline method argument `$lead` voluntary
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
There is already an API endpoint to get events for 1 or limited number of contacts, but there was none for getting contact events unrelated to contacts.

Note: Some events like `lead.identified` or `lead.create` will not return any events since those events require a lead object. The functionality can be added though.

#### Steps to test this PR:
1. Apply this PR
2. Clear the cache
3. Apply https://github.com/mautic/api-library/pull/126 to your API Library
4. Run `phpunit --filter testGetAllEvents`
5. Try to access the API endpoint with some filters with your favorite API testing app and make sure the results are as you'd expect based on your filters. Use the attached docs to know all possibilities.
6. Make sure the timeline on a contact detail page still works as before.
